File: example/src/main/java/eu/darken/mvpbakery/example/screens/counting/CountingFragment.java
package eu.darken.mvpbakery.example.screens.counting;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;
import eu.darken.mvpbakery.base.MVPBakery;
import eu.darken.mvpbakery.base.StateForwarder;
import eu.darken.mvpbakery.base.ViewModelRetainer;
import eu.darken.mvpbakery.example.R;
import eu.darken.mvpbakery.injection.InjectedPresenter;
import eu.darken.mvpbakery.injection.PresenterInjectionCallback;

public class CountingFragment extends Fragment implements CountingPresenter.View {

    @BindView(R.id.fragment_text) TextView textView;

    private StateForwarder stateForwarder;
    private MVPBakery<CountingPresenter.View, CountingPresenter> mvpBakery;
    private Unbinder unbinder;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        stateForwarder = new StateForwarder();
        stateForwarder.onCreate(savedInstanceState);
        mvpBakery = MVPBakery.<CountingPresenter.View, CountingPresenter>builder()
                .stateForwarder(stateForwarder)
                .presenterFactory(new InjectedPresenter<>(this))
                .presenterRetainer(new ViewModelRetainer<>(this))
                .addPresenterCallback(new PresenterInjectionCallback<>(this))
                .attach(this);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        final View view = inflater.inflate(R.layout.fragment_counting, container, false);
        unbinder = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onDestroyView() {
        unbinder.unbind();
        super.onDestroyView();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        stateForwarder.onSaveInstanceState(outState);
        super.onSaveInstanceState(outState);
    }

    @Override
    public void showText(String text) {
        textView.setText(text);
    }

    @OnClick(R.id.fragment_button)
    void onCountClick() {
        mvpBakery.getPresenter().onCountClick();
    }
}

File: example/src/main/java/eu/darken/mvpbakery/example/screens/counting/CountingFragment.kt
package eu.darken.mvpbakery.example.screens.counting

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import butterknife.Unbinder
import eu.darken.mvpbakery.base.MVPBakery
import eu.darken.mvpbakery.base.StateForwarder
import eu.darken.mvpbakery.base.ViewModelRetainer
import eu.darken.mvpbakery.example.R
import eu.darken.mvpbakery.injection.InjectedPresenter
import eu.darken.mvpbakery.injection.PresenterInjectionCallback

class CountingFragment : Fragment(), CountingPresenter.View {

    @BindView(R.id.fragment_text)
    lateinit var textView: TextView
    @BindView(R.id.presenter_info)
    lateinit var presenterInfo: TextView

    private lateinit var stateForwarder: StateForwarder
    private lateinit var mvpBakery: MVPBakery<CountingPresenter.View, CountingPresenter>
    private var unbinder: Unbinder? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        stateForwarder = StateForwarder()
        stateForwarder.onCreate(savedInstanceState)
        mvpBakery = MVPBakery.builder<CountingPresenter.View, CountingPresenter>()
                .stateForwarder(stateForwarder)
                .presenterFactory(InjectedPresenter(this))
                .presenterRetainer(ViewModelRetainer(this))
                .addPresenterCallback(PresenterInjectionCallback(this))
                .attach(this)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_counting, container, false)
        unbinder = ButterKnife.bind(this, view)
        return view
    }

    override fun onDestroyView() {
        unbinder!!.unbind()
        super.onDestroyView()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        stateForwarder.onSaveInstanceState(outState)
        super.onSaveInstanceState(outState)
    }

    override fun showText(text: String) {
        textView.text = text
    }

    @OnClick(R.id.fragment_button)
    internal fun onCountClick() {
        mvpBakery.presenter!!.onCountClick()
    }

    override fun showPresenterInfo(presenter: CountingPresenter) {
        presenterInfo.text = presenter.toString()
    }
}
